trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/plugin/TrinoAwsProxyServerPlugin.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.spi.plugin;
+
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+import io.trino.aws.proxy.spi.credentials.AssumedRoleProvider;
+import io.trino.aws.proxy.spi.credentials.CredentialsProvider;
+import io.trino.aws.proxy.spi.plugin.config.AssumedRoleProviderConfig;
+import io.trino.aws.proxy.spi.plugin.config.CredentialsProviderConfig;
+import io.trino.aws.proxy.spi.plugin.config.PluginIdentifierConfig;
+import io.trino.aws.proxy.spi.plugin.config.PluginOptionalIdentifierConfig;
+import io.trino.aws.proxy.spi.plugin.config.S3DatabaseSecurityFacadeProviderConfig;
+import io.trino.aws.proxy.spi.plugin.config.S3SecurityFacadeProviderConfig;
+import io.trino.aws.proxy.spi.security.S3DatabaseSecurityFacadeProvider;
+import io.trino.aws.proxy.spi.security.S3SecurityFacadeProvider;
+
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static com.google.inject.util.Modules.combine;
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
+
+public interface TrinoAwsProxyServerPlugin
```

Comment:
Maybe for a separate PR... we should have an example Plugin for testing purposes.

Reply:
Agree, will look into it for a future PR